bundles/org.openhab.binding.wundergroundupdatereceiver/src/main/resources/OH-INF/thing/channel-group-types.xml

```diff
@@ -0,0 +1,118 @@
+<?xml version="1.0" ?>
+<thing:thing-descriptions bindingId="wundergroundupdatereceiver"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="https://openhab.org/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/thing-description/v1.0.0 https://openhab.org/schemas/thing-description-1.0.0.xsd">
+
+	<channel-group-type id="metadata-group">
+		<label>Metadata</label>
+		<description>Metadata regarding the last submission</description>
+		<category>Text</category>
+		<channels>
+			<channel id="last-received" typeId="last-received-datetime"/>
+			<channel id="dateutc" typeId="datetime-utc"/>
+			<channel id="softwaretype" typeId="softwaretype"/>
+			<channel id="last-query-state" typeId="last-query-state"/>
+			<channel id="last-query-trigger" typeId="last-query-trigger"/>
+		</channels>
+	</channel-group-type>
+
+	<channel-group-type id="wind-group">
+		<label>Wind</label>
+		<description>Wind measurements</description>
+		<category>Wind</category>
+		<channels>
+			<channel id="winddir" typeId="system.wind-direction"/>
```

Comment:
can you add winddir as text as well?
`["N", "NNO", "NO", "ONO", "O", "OSO", "SO", "SSO", "S", "SSW", "SW", "WSW", "W", "WNW", "NW", "NNW"]`

Reply:
I would do that in a rule.

Reply:
You may want to provide the following example JS transformation in the README

```
(function(s){ 
  if ( (s == "NULL") || (s == "UNDEF") )
  {
      return undefined;
  }
  else
  {
    var dir = ["N ⬆️", "NNO ⬆️", "NO ↗️", "ONO ➡️", "O ➡️", "OSO ➡️", "SO ↘️", "SSO ⬇️", "S ⬇️", "SSW ⬇️", "SW ↙️", "WSW ⬅️", "W ⬅️", "WNW ⬅️", "NW ↖️", "NNW ⬆️"];
    var wind          = parseInt(s.split(" ")[0]);
    var winddiroffset = (wind + (360.0/32.0)) % 360.0;
    var winddiridx    = Math.floor(winddiroffset / (360.0/16.0));
    var winddir       = dir[winddiridx];
    
    return winddir + ' ('+ wind +'°)';
  }
})(input)
```

Reply:
May I ask how you use such a transformation? There are vast swaths of things I don't know about openhab 3.

Reply:
1. Install Javascript Transformation
2. save script from above under `openhab/conf/transform/winddir.js`
3. use script via Item -> _Add Metadata: stateDescription_ -> _pattern:_ `JS(winddir.js):%s`

Reply:
Please change ...
`var dir = ["N ⬆️", "NNO ⬆️", "NO ↗️", "ONO ➡️", "O ➡️", "OSO ➡️", "SO ↘️", "SSO ⬇️", "S ⬇️", "SSW ⬇️", "SW ↙️", "WSW ⬅️", "W ⬅️", "WNW ⬅️", "NW ↖️", "NNW ⬆️"];`
... to ...
`var dir = ["N ⬇️", "NNO ⬇️", "NO ↙️", "ONO ⬅️", "O ⬅️", "OSO ⬅️", "SO ↖️", "SSO ⬆️", "S ⬆️", "SSW ⬆️", "SW ↗️", "WSW ➡️", "W ➡️", "WNW ➡️", "NW ↘️", "NNW ⬇️"];`

The arrows need to point to the direction where the wind is going to (and not coming from).